memory_quota: tidb bootstrap failed

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
topology.yaml
```
global:
  user: "root"
  ssh_port: 22
  deploy_dir: "/tidb-deploy"
  data_dir: "/tidb-data"

pd_servers:
  - host: 172.16.4.57
    client_port: 2379
    peer_port: 2380
tidb_servers:
  - host: 172.16.4.57
    port: 4000
    status_port: 10080
    config:
      tikv-client.store-liveness-timeout: "5s"
      oom-use-tmp-storage: false
      oom-action: log
      mem-quota-query: 1
tikv_servers:
  - host: 172.16.4.57
    port: 20160
    status_port: 20180
```
tiup cluster deploy -p cp-test v4.0.2 topology.yaml
tiup cluster start cp-test
```
[root@localhost ~]# tiup cluster start cp-test
Starting component `cluster`: /root/.tiup/components/cluster/v1.0.5/tiup-cluster start cp-test
Starting cluster cp-test...
+ [ Serial ] - SSHKeySet: privateKey=/root/.tiup/storage/cluster/clusters/cp-test/ssh/id_rsa, publicKey=/root/.tiup/storage/cluster/clusters/cp-test/ssh/id_rsa.pub
+ [Parallel] - UserSSH: user=root, host=172.16.4.57
+ [Parallel] - UserSSH: user=root, host=172.16.4.57
+ [Parallel] - UserSSH: user=root, host=172.16.4.57
+ [ Serial ] - ClusterOperate: operation=StartOperation, options={Roles:[] Nodes:[] Force:false SSHTimeout:5 OptTimeout:60 APITimeout:300}
Starting component pd
	Starting instance pd 172.16.4.57:2379
	Start pd 172.16.4.57:2379 success
Starting component node_exporter
	Starting instance 172.16.4.57
	Start 172.16.4.57 success
Starting component blackbox_exporter
	Starting instance 172.16.4.57
	Start 172.16.4.57 success
Starting component tikv
	Starting instance tikv 172.16.4.57:20160
	Start tikv 172.16.4.57:20160 success
Starting component tidb
	Starting instance tidb 172.16.4.57:4000
retry error: operation timed out after 1m0s
	tidb 172.16.4.57:4000 failed to start: timed out waiting for port 4000 to be started after 1m0s, please check the log of the instance

Error: failed to start: failed to start tidb: 	tidb 172.16.4.57:4000 failed to start: timed out waiting for port 4000 to be started after 1m0s, please check the log of the instance: timed out waiting for port 4000 to be started after 1m0s

Verbose debug logs has been written to /root/logs/tiup-cluster-debug-2020-07-07-01-25-59.log.
Error: run `/root/.tiup/components/cluster/v1.0.5/tiup-cluster` (wd:/root/.tiup/data/S40wb1p) failed: exit status 1
```

tidb_stderr.log
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2217edd]

goroutine 1 [running]:
github.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1(0xc000de7400, 0xc000c74960, 0x37194c0, 0xc0000e0000)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/adapter.go:291 +0x53a
panic(0x2e32d80, 0x4cf30c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb/util/expensivequery.(*Handle).LogOnQueryExceedMemQuota(0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/util/expensivequery/expensivequery.go:91 +0x4d
github.com/pingcap/tidb/util/memory.(*LogOnExceed).Action(0xc000e5fe00, 0xc000e62c40)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/util/memory/action.go:64 +0xbd
github.com/pingcap/tidb/util/memory.(*Tracker).Consume(0xc000e62d20, 0xa73)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/util/memory/tracker.go:221 +0x118
github.com/pingcap/tidb/executor.insertRows(0x37194c0, 0xc0000e0000, 0x36e9a80, 0xc000e5b980, 0x0, 0x2fcdc00)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/insert_common.go:249 +0x471
github.com/pingcap/tidb/executor.(*InsertExec).Next(0xc000e5b980, 0x37194c0, 0xc0000e0000, 0xc000e51130, 0x308ac00, 0xc000384380)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/insert.go:262 +0x73
github.com/pingcap/tidb/executor.Next(0x37194c0, 0xc0000e0000, 0x3722cc0, 0xc000e5b980, 0xc000e51130, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/executor.go:248 +0x11f
github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor(0xc000de7400, 0x37194c0, 0xc0000e0000, 0x3722cc0, 0xc000e5b980, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/adapter.go:506 +0x2d6
github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay(0xc000de7400, 0x37194c0, 0xc0000e0000, 0x3722cc0, 0xc000e5b980, 0x5210300, 0xc000c74b00, 0x0, 0xc000c747f0, 0x116c987, ...)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/adapter.go:388 +0x13a
github.com/pingcap/tidb/executor.(*ExecStmt).Exec(0xc000de7400, 0x37194c0, 0xc0000e0000, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/executor/adapter.go:356 +0x3c7
github.com/pingcap/tidb/session.runStmt(0x37194c0, 0xc0000e0000, 0x375c4a0, 0xc000a59100, 0x3722980, 0xc000de7400, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/tidb.go:276 +0x304
github.com/pingcap/tidb/session.(*session).executeStatement(0xc000a59100, 0x37194c0, 0xc0000e0000, 0x0, 0x3721e80, 0xc000e68680, 0x3722980, 0xc000de7400, 0x0, 0x0, ...)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/session.go:1038 +0xb5
github.com/pingcap/tidb/session.(*session).execute(0xc000a59100, 0x37194c0, 0xc0000e0000, 0x3266cdd, 0xd5, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/session.go:1151 +0x78d
github.com/pingcap/tidb/session.(*session).Execute(0xc000a59100, 0x37194c0, 0xc0000e0000, 0x3266cdd, 0xd5, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/session.go:1081 +0xdb
github.com/pingcap/tidb/session.mustExecute(0x37915e0, 0xc000a59100, 0x3266cdd, 0xd5)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/bootstrap.go:1203 +0x7e
github.com/pingcap/tidb/session.doDMLWorks(0x37915e0, 0xc000a59100)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/bootstrap.go:1145 +0x98
github.com/pingcap/tidb/session.bootstrap(0x37915e0, 0xc000a59100)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/bootstrap.go:311 +0x359
github.com/pingcap/tidb/session.runInBootstrapSession(0x374c3e0, 0xc00087c100, 0x329a948)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/session.go:1759 +0x266
github.com/pingcap/tidb/session.BootstrapSession(0x374c3e0, 0xc00087c100, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/session/session.go:1653 +0x89f
main.createStoreAndDomain()
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:295 +0x193
main.main()
	/home/jenkins/agent/workspace/tidb_v4.0.2/go/src/github.com/pingcap/tidb/tidb-server/main.go:181 +0x1cd
```
### 2. What did you expect to see? (Required)
TiDB bootstrap success

### 3. What did you see instead (Required)
TiDB bootstrap failed

### 4. Affected version (Required)

<!-- v3.0.0, v4.0.0, etc -->
4.0.2
### 5. Root Cause Analysis

<!-- should be filled by the investigator before it's closed -->

--- a/domain/domain.go
+++ b/domain/domain.go
@@ -1297,11 +1297,6 @@ func (do *Domain) ExpensiveQueryHandle() *expensivequery.Handle {
 	return do.expensiveQueryHandle
 }
 
-// InitExpensiveQueryHandle init the expensive query handler.
-func (do *Domain) InitExpensiveQueryHandle() {
-	do.expensiveQueryHandle = expensivequery.NewExpensiveQueryHandle(do.exit)
-}
-
 const privilegeKey = "/tidb/privilege"
 
 // NotifyUpdatePrivilege updates privilege key in etcd, TiDB client that watches

--- a/domain/domain.go
+++ b/domain/domain.go
@@ -692,6 +692,7 @@ func NewDomain(store kv.Storage, ddlLease time.Duration, statsLease time.Duratio
 	}
 
 	do.SchemaValidator = NewSchemaValidator(ddlLease, do)
+	do.expensiveQueryHandle = expensivequery.NewExpensiveQueryHandle(do.exit)
 	return do
 }
 

--- a/session/bootstrap_test.go
+++ b/session/bootstrap_test.go
@@ -475,19 +475,6 @@ func (s *testBootstrapSuite) TestOldPasswordUpgrade(c *C) {
 	c.Assert(newpwd, Equals, "*0D3CED9BEC10A777AEC23CCC353A8C08A633045E")
 }
 
-func (s *testBootstrapSuite) TestBootstrapInitExpensiveQueryHandle(c *C) {
-	defer testleak.AfterTest(c)()
-	store := newStore(c, s.dbName)
-	defer store.Close()
-	se, err := createSession(store)
-	c.Assert(err, IsNil)
-	dom := domain.GetDomain(se)
-	c.Assert(dom, NotNil)
-	defer dom.Close()
-	dom.InitExpensiveQueryHandle()
-	c.Assert(dom.ExpensiveQueryHandle(), NotNil)
-}
-
 func (s *testBootstrapSuite) TestStmtSummary(c *C) {
 	defer testleak.AfterTest(c)()
 	ctx := context.Background()

--- a/session/bootstrap_test.go
+++ b/session/bootstrap_test.go
@@ -475,6 +475,18 @@ func (s *testBootstrapSuite) TestOldPasswordUpgrade(c *C) {
 	c.Assert(newpwd, Equals, "*0D3CED9BEC10A777AEC23CCC353A8C08A633045E")
 }
 
+func (s *testBootstrapSuite) TestBootstrapInitExpensiveQueryHandle(c *C) {
+	defer testleak.AfterTest(c)()
+	store := newStore(c, s.dbName)
+	defer store.Close()
+	se, err := createSession(store)
+	c.Assert(err, IsNil)
+	dom := domain.GetDomain(se)
+	c.Assert(dom, NotNil)
+	defer dom.Close()
+	c.Assert(dom.ExpensiveQueryHandle(), NotNil)
+}
+
 func (s *testBootstrapSuite) TestStmtSummary(c *C) {
 	defer testleak.AfterTest(c)()
 	ctx := context.Background()

--- a/session/session.go
+++ b/session/session.go
@@ -2172,7 +2172,6 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
 	}
 
 	dom := domain.GetDomain(se)
-	dom.InitExpensiveQueryHandle()
 
 	se2, err := createSession(store)
 	if err != nil {

